cmake/catkin_libraries.cmake

```diff
@@ -131,8 +131,15 @@ function(catkin_replace_imported_library_targets VAR)
       # sometimes cmake dependencies define imported targets, in which
       # case the imported library information is not the target name, but
       # the information embedded in cmake properties inside the imported library
+      # For interface libraries, this is the INTERFACE_LINK_LIBRARIES property.
+      # For regular imported libraries, this is IMPORTED_LOCATION(_$CONFIG).
+      get_target_property(${lib}_type ${lib} TYPE)
       get_target_property(${lib}_imported ${lib} IMPORTED)
-      if(${${lib}_imported})
+      if(${${lib}_type} STREQUAL "INTERFACE_LIBRARY")
```

Comment:
I couldn't find any documentation about the "string" `INTERFACE_LIBRARY`.

Maybe `get_property(... SET/DEFINED)` could be used instead if this can't be relied on?

Reply:
Documentation is here: https://cmake.org/cmake/help/v3.1/prop_tgt/TYPE.html

It first appeared in cmake 3.1, but since older cmakes will never report `"INTERFACE_LIBRARY"`, it is automatically backwards compatible.

Reply:
Ok, thanks for the reference. I was indeed looking at the 3.0 docs.